docgen/docgen.go

```diff
@@ -52,18 +53,33 @@ func (p *docPlugin) Generate(req *plugin_go.CodeGeneratorRequest) (*plugin_go.Co
 		}
 	}
 
-	var source *google_protobuf.FileDescriptorProto
+	var source *parser.FileDescWrapper
 	for _, f := range req.GetProtoFile() {
 		if strings.Contains(f.GetName(), "all.proto") {
-			source = f
-			break
+			for _, fileToGenerate := range req.FileToGenerate {
+				if fileToGenerate == f.GetName() {
+					source = &parser.FileDescWrapper{FileDescriptorProto: f}
+					break
+				}
+			}
 		}
 	}
 
 	if source == nil {
 		return nil, fmt.Errorf("no file to generate")
 	}
 
+	source.DependencyMap = make(map[string]*google_protobuf.FileDescriptorProto)
+
+	// generate dependency array for the source
```

Comment:
sorry, what do you mean by "array"? I just see a map :)

Reply:
changed the comment